### PR TITLE
Split state in 4 smaller ones

### PR DESCRIPTION
**Pull request by Quetzacoalt91**
- [x] :warning: This PR is based on https://github.com/PrestaShop/autoupgrade/pull/1076, and needs this one to be merged first.

| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | The state contains some data to keep track of the current running process. It contains some data only useful to the update, some other data for the restore and finally the rest for the backup. The execution of a process may impact the data of another process (for instance, flagging a backup has been completed right before the update), but the State class is always deleted at the end of this given process. To avoid side effect in the future, we split the current state in 3 smaller ones, and introduce at the same time a new one containing the most recent log files for each process.
| Type?             | refactor
| BC breaks?        | Yes, some unused methods and logic have been removed (i.e setTableLines)
| Deprecations?     | Yes: `warning_exist` will be deleted as it is unused on the incoming UI with the v7.
| Fixed ticket?     | /
| Sponsor company   | @PrestaShopCorp
| How to test?      | After each process (backup and update at the moment), the related state file is properly deleted (state_backup.var / state_update.var), the state_logs.var file is still available in `admin_folder/autoupgrade` and the logs are downloadable from the web interface.

### CODE REVIEW

File: classes/Parameters/UpgradeFileNames.php
Comment (ga-devfront) [inline classes/Parameters/UpgradeFileNames.php]:
```suggestion
```

This variable does not seem to be called in the project, I think we can remove it.

File: classes/Task/AbstractTask.php
Comment (Quetzacoalt91) [inline classes/Task/AbstractTask.php]: Duplicated initialization of restore state with task Backup inititialization ?

File: classes/Task/Backup/BackupFiles.php
Comment (ga-devfront) [inline classes/Task/Backup/BackupFiles.php]: I think you can add a variable to not reuse `$this->container->getBackupState()` all the time but just `$backupState` or `$state`

File: classes/Task/Restore/Restore.php
Comment (ga-devfront) [inline classes/Task/Restore/Restore.php]: I think you can variabilise the state again `$state = $this->container->getRestoreState()` instead of reusing all the time `$this->container->getRestoreState()`

File: classes/Task/Restore/RestoreFiles.php
Comment (ga-devfront) [inline classes/Task/Restore/RestoreFiles.php:51]: same for variable `$state`

File: classes/Task/Update/UpdateComplete.php
Comment (ga-devfront) [inline classes/Task/Update/UpdateComplete.php]: you can variable it again `$state = $this->container->getUpdateState()`

File: classes/Task/Update/UpdateDatabase.php
Comment (ga-devfront) [inline classes/Task/Update/UpdateDatabase.php:56]: and again

File: classes/Task/Update/UpdateFiles.php
Comment (ga-devfront) [inline classes/Task/Update/UpdateFiles.php:88]: one more time baby

**Review by ga-devfront — CHANGES_REQUESTED** [review]
image

File: classes/Task/Update/UpdateDatabase.php
Comment (Quetzacoalt91) [reply classes/Task/Update/UpdateDatabase.php:56]: This file has one call to the state per method, storing it in a variable is not really important.

File: classes/Task/Update/UpdateFiles.php
Comment (Quetzacoalt91) [reply classes/Task/Update/UpdateFiles.php:88]: Done in `warmUp` method but not in `run()` as there is only one occurrence of the state in it.

File: classes/Task/Restore/RestoreFiles.php
Comment (Quetzacoalt91) [reply classes/Task/Restore/RestoreFiles.php:51]: Done, even though "Outdated" does not appear here.

**Review by ga-devfront — APPROVED** [review]

**Comment by ps-jarvis** [comment]
PR merged, well done!

Message to @PrestaShop/committers: do not forget to milestone it before the merge.

**Pull request merged**
